src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -323,9 +324,13 @@ typedef struct {
   bool insert_null_before_wire_value;
 } wire_value;
 
+template <bool mdkey_definitely_interned>
```

Comment:
Mostly to avoid typing up two almost identical methods. 

If we kept it as a single method that takes mdkey_definitely_interned as a parameter, we introduce an unnecessary branch.

Reply:
Note that by adding templates, we are also increasing code size. At some point, there will be a tradeoff where we need to keep in mind the cache misses due to increased code size.